### PR TITLE
Document YAXConfig

### PR DESCRIPTION
**Pull request by gdkrmr**
I have added some documentation to YAXConfig. There are two fields that I don't know what they are doing, maybe someone can add something there.

### CODE REVIEW

File: src/YAXArrays.jl [hunk 1]
```diff
@@ -1,4 +1,15 @@
 module YAXArrays
+
+"""
+Default configuration for YAXArrays, has the following fields:
+
+- `workdir[]::String = "./"` The default location for temporary cubes.
+- `recal[]::Bool = false` set to true if you want `@loadOrGenerate` to always recalculate the results.
+- `chunksize[]::Any = :input` TODO
```
Comment (felixcremer) [inline src/YAXArrays.jl:8]:
```suggestion
- `chunksize[]::Any = :input` Set the default output chunksize.
```

File: src/YAXArrays.jl [hunk 2]
```diff
@@ -1,4 +1,15 @@
...
+- `max_cache[]::Float64 = 1e8` The maximum cache used by mapCube.
+- `cubedir[]::""` the default location for `Cube()` without an argument.
+- `subsetextension::Array{Any} = []` TODO
```
Comment (felixcremer) [inline src/YAXArrays.jl:11]:
```suggestion
- `subsetextensions::Array{Any} = []` List of registered functions, that convert subsetting input into dimension boundaries. 
```

Comment (felixcremer) [reply src/YAXArrays.jl:11]: This is used in EarthDataLab to register the replaceregion function to enable spatial subsampling with region names. 
https://github.com/JuliaDataCubes/EarthDataLab.jl/blob/ee0e13f7b5560d0d8638c2e20fcfd24281393b46/src/EarthDataLab.jl#L39
I am not sure, whether we should add this example.

**Comment by gdkrmr** [comment]
Thanks for the corrections. I'll wait for the tests and then I just merge!

**Comment by felixcremer** [comment]
Thanks.

**Comment by gdkrmr** [comment]
wtf. I pressed Ctrl-Enter to send the comment and it just merged everything .... hmmmm :-/

**Comment by gdkrmr** [comment]
@felixcremer did you merge this?

**Comment by felixcremer** [comment]
Yes I went ahead and merged it.

**Pull request merged**